common/rya.api/src/main/java/org/apache/rya/api/RdfCloudTripleStoreConfigurationBuilder.java

```diff
@@ -0,0 +1,158 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.rya.api;
+
+import com.google.common.base.Preconditions;
+
+/**
+ * This is a base class meant to be extended by Rya configuration builders. Any
+ * class extending this class inherits setter methods that should be common to
+ * all Rya implementations.
+ * 
+ * @param <B>
+ *            the configuration builder returned by each of the setter methods
+ * @param <C>
+ *            the configuration object returned by the builder when build is
+ *            called
+ */
+public abstract class RdfCloudTripleStoreConfigurationBuilder<B extends RdfCloudTripleStoreConfigurationBuilder<B, C>, C extends RdfCloudTripleStoreConfiguration> {
+
+    private String prefix = "rya_";
+    private String auths;
+    private String visibilities;
+    private boolean useInference = false;
+    private boolean displayPlan = false;
+
+    /**
+     * Auxiliary method to return specified Builder in a type safe way
+     * 
+     * @return specified builder
+     */
+    protected abstract B confBuilder();
+
+    /**
+     * Auxiliary method to return type of configuration object constructed by
+     * builder in a type safe way
+     * 
+     * @return specified configuration
+     */
+    protected abstract C createConf();
+
+    /**
+     * Set whether to use backwards chaining inferencing during query
+     * evaluation. The default value is false.
+     * 
+     * @param useInference
+     *            - turn inferencing on and off in Rya
+     * @return B - concrete builder class
+     */
+    public B useInference(boolean useInference) {
+        this.useInference = useInference;
+        return confBuilder();
+    }
+
+    /**
+     * 
+     * Sets the authorization for querying the underlying data store.
+     * 
+     * @param auths
+     *            - authorizations for querying underlying datastore
+     * @return B - concrete builder class
+     */
+    public B auths(String auths) {
+        Preconditions.checkNotNull(auths, "User auths cannot be null.");
+        this.auths = auths;
+        return confBuilder();
+    }
+
+    /**
+     * Set the column visibities for ingested triples. If no value is set,
+     * triples won't have a visibility.
+     * 
+     * @param visibilities
+     *            - visibilities assigned to any triples inserted into Rya
+     * @return B - concrete builder class
+     */
+    public B visibilities(String visibilites) {
+        Preconditions.checkNotNull(visibilites, "Visibilities cannot be null.");
+        this.visibilities = visibilites;
+        return confBuilder();
+    }
+
+    /**
+     *
+     * Sets the prefix for the Rya instance to connect to. This parameter is
+     * required and the default value is "rya_"
+     * 
+     * @param prefix
+     *            - the prefix for the Rya instance
+     * @return B - concrete builder class
+     */
+    public B setRyaPrefix(String prefix) {
```

Comment:
You should use a consistent method naming paradigm within your class. I think this is the only setter that says "set" on the start. Either have all of them say set, or none of them. This might also be a good opportunity to change "rya prefix" to "rya instance name" since the prefix is Accumulo specific.